ingest/indexer/service/indexer_streaming_service.go

```diff
@@ -240,6 +245,43 @@ func (s *indexerStreamingService) publishTxn(ctx context.Context, req types.Requ
 	return s.client.PublishTransaction(sdkCtx, txn)
 }
 
+// addTokenLiquidity adds the token liquidity to the event.
+// It refers to the pooled amount of each asset after a swap event has occurred.
+func (s *indexerStreamingService) AddTokenLiquidity(ctx context.Context, event *types.Event) error {
```

Comment:
I think that it would be good to return a new deep copy of the event rather than mutating the given one for the state-breaking nature of events and possible edge cases around it that we can derisk with a deep copy.

Reply:
Done as part of the deepCloneEvent function